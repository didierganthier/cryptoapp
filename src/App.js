import React from "react";
import { Cryptocurrencies, CryptoDetails, Exchanges, Homepage, Navbar, News } from "./components";
import './App.css';
import { Layout, Space, Typography } from "antd";
import { Link, Route, Routes } from "react-router-dom";

const App = () => {
    return (
        <div className="app">
            <div className="navbar">
                <Navbar />
            </div>
            <div className="main">
                <Layout>
                    <div className="routes">
                        <Routes>
                            <Route exact path="/" element={<Homepage />} />
                            <Route exact path="/exchanges" element={<Exchanges />} />
                            <Route exact path="/cryptocurrencies" element={<Cryptocurrencies />} />
                            <Route exact path="/crypto/:coinId" element={<CryptoDetails />} />
                            <Route exact path="/news" element={<News />} />
                        </Routes>
                    </div>
                </Layout>
                <div className="footer">
                    <Typography.Title level={5} style={{ color: 'white', textAlign: 'center' }}>Copyright Â© 2021
                        <Link to="/">
                            Cryptoverse Inc.
                        </Link> <br />
                        All Rights Reserved.
                    </Typography.Title>
                    <Space>
                        <Link to="/">Home</Link>
                        <Link to="/exchanges">Exchanges</Link>
                        <Link to="/news">News</Link>
                    </Space>
                </div>
            </div>
        </div>
    )
}

export default App
