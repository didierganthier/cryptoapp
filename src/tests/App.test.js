// Render app
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import 'antd/dist/antd';
import { render } from '@testing-library/react';
import App from '../App';
import store from '../app/store';

// Write tests here
describe('App', () => {
  it('renders without crashing', () => {
    window.matchMedia = window.matchMedia || function () {
      return {
        matches: false,
        addListener: () => {},
        removeListener: () => {},
      };
    };
    const tree = render(
      <Router>
        <Provider store={store}>
          <App />
        </Provider>
      </Router>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('contains a navbar', () => {
    const { getByText } = render(
      <Router>
        <Provider store={store}>
          <App />
        </Provider>
      </Router>,
    );
    expect(getByText('Cryptoverse')).toBeInTheDocument();
  });
});
