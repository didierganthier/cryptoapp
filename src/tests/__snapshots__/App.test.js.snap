// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="bg-white border-gray-200 px-2 sm:px-4 py-2.5 rounded-b dark:bg-gray-900"
        >
          <div
            class="container flex flex-wrap items-center justify-between mx-auto"
          >
            <a
              class="flex items-center"
              href="/"
            >
              <img
                alt="Flowbite Logo"
                class="h-6 mr-3 sm:h-9"
                src="https://flowbite.com/docs/images/logo.svg"
              />
              <span
                class="self-center text-xl font-semibold whitespace-nowrap dark:text-white"
              >
                Cryptoverse
              </span>
            </a>
            <button
              aria-controls="navbar-default"
              aria-expanded="false"
              class="inline-flex items-center p-2 ml-3 text-sm text-gray-500 rounded-lg md:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600"
              data-collapse-toggle="navbar-default"
              type="button"
            >
              <span
                class="sr-only"
              >
                Open main menu
              </span>
              <svg
                aria-hidden="true"
                class="w-6 h-6"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
            <div
              class="hidden w-full md:block md:w-auto"
              id="navbar-default"
            >
              <ul
                class="flex flex-col p-4 mt-4 border border-gray-100 rounded-lg bg-gray-50 md:flex-row md:space-x-8 md:mt-0 md:text-sm md:font-medium md:border-0 md:bg-white dark:bg-gray-800 md:dark:bg-gray-900 dark:border-gray-700"
              >
                <li>
                  <a
                    class="block py-2 pl-3 pr-4 text-gray-700 rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-blue-700 md:p-0 dark:text-gray-400 md:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white md:dark:hover:bg-transparent"
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li>
                  <a
                    class="block py-2 pl-3 pr-4 text-gray-700 rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-blue-700 md:p-0 dark:text-gray-400 md:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white md:dark:hover:bg-transparent"
                    href="/cryptocurrencies"
                  >
                    Cryptocurrencies
                  </a>
                </li>
                <li>
                  <a
                    class="block py-2 pl-3 pr-4 text-gray-700 rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-blue-700 md:p-0 dark:text-gray-400 md:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white md:dark:hover:bg-transparent"
                    href="/news"
                  >
                    News
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </nav>
        <div
          class="carousel slide relative"
          data-bs-ride="carousel"
          id="carouselExampleSlidesOnly"
        >
          <div
            class="carousel-inner relative md:h-[600px] h-[300px] w-full overflow-hidden"
          >
            <div
              class="carousel-item active relative float-left w-full"
            >
              <img
                alt="Wild Landscape"
                class="block h-1/2 w-full"
                src="https://www.interactivebrokers.hu/images/web/cryptocurrency-hero.jpg"
              />
            </div>
          </div>
        </div>
        <div
          class="app"
        >
          <div
            class="main"
          >
            <section
              class="ant-layout css-dev-only-do-not-override-8jqkd1"
            >
              <div
                class="routes"
              >
                <div
                  class="loader"
                >
                  <div
                    aria-busy="true"
                    aria-live="polite"
                    class="ant-spin ant-spin-spinning css-dev-only-do-not-override-8jqkd1"
                  >
                    <span
                      class="ant-spin-dot ant-spin-dot-spin"
                    >
                      <i
                        class="ant-spin-dot-item"
                      />
                      <i
                        class="ant-spin-dot-item"
                      />
                      <i
                        class="ant-spin-dot-item"
                      />
                      <i
                        class="ant-spin-dot-item"
                      />
                    </span>
                  </div>
                </div>
              </div>
            </section>
            <div
              class="footer"
            >
              <h5
                class="ant-typography css-dev-only-do-not-override-8jqkd1"
                style="color: white; text-align: center;"
              >
                Copyright Â© 2021
                <a
                  href="/"
                >
                  Cryptoverse Inc.
                </a>
                 
                <br />
                <p>
                  All Rights Reserved.
                </p>
              </h5>
              <div
                class="ant-space css-dev-only-do-not-override-8jqkd1 ant-space-horizontal ant-space-align-center"
              >
                <div
                  class="ant-space-item"
                  style="margin-right: 8px;"
                >
                  <a
                    href="/"
                  >
                    Home
                  </a>
                </div>
                <div
                  class="ant-space-item"
                  style="margin-right: 8px;"
                >
                  <a
                    href="/cryptocurrencies"
                  >
                    Cryptocurrencies
                  </a>
                </div>
                <div
                  class="ant-space-item"
                >
                  <a
                    href="/news"
                  >
                    News
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="bg-white border-gray-200 px-2 sm:px-4 py-2.5 rounded-b dark:bg-gray-900"
      >
        <div
          class="container flex flex-wrap items-center justify-between mx-auto"
        >
          <a
            class="flex items-center"
            href="/"
          >
            <img
              alt="Flowbite Logo"
              class="h-6 mr-3 sm:h-9"
              src="https://flowbite.com/docs/images/logo.svg"
            />
            <span
              class="self-center text-xl font-semibold whitespace-nowrap dark:text-white"
            >
              Cryptoverse
            </span>
          </a>
          <button
            aria-controls="navbar-default"
            aria-expanded="false"
            class="inline-flex items-center p-2 ml-3 text-sm text-gray-500 rounded-lg md:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600"
            data-collapse-toggle="navbar-default"
            type="button"
          >
            <span
              class="sr-only"
            >
              Open main menu
            </span>
            <svg
              aria-hidden="true"
              class="w-6 h-6"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <div
            class="hidden w-full md:block md:w-auto"
            id="navbar-default"
          >
            <ul
              class="flex flex-col p-4 mt-4 border border-gray-100 rounded-lg bg-gray-50 md:flex-row md:space-x-8 md:mt-0 md:text-sm md:font-medium md:border-0 md:bg-white dark:bg-gray-800 md:dark:bg-gray-900 dark:border-gray-700"
            >
              <li>
                <a
                  class="block py-2 pl-3 pr-4 text-gray-700 rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-blue-700 md:p-0 dark:text-gray-400 md:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white md:dark:hover:bg-transparent"
                  href="/"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  class="block py-2 pl-3 pr-4 text-gray-700 rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-blue-700 md:p-0 dark:text-gray-400 md:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white md:dark:hover:bg-transparent"
                  href="/cryptocurrencies"
                >
                  Cryptocurrencies
                </a>
              </li>
              <li>
                <a
                  class="block py-2 pl-3 pr-4 text-gray-700 rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-blue-700 md:p-0 dark:text-gray-400 md:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white md:dark:hover:bg-transparent"
                  href="/news"
                >
                  News
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
      <div
        class="carousel slide relative"
        data-bs-ride="carousel"
        id="carouselExampleSlidesOnly"
      >
        <div
          class="carousel-inner relative md:h-[600px] h-[300px] w-full overflow-hidden"
        >
          <div
            class="carousel-item active relative float-left w-full"
          >
            <img
              alt="Wild Landscape"
              class="block h-1/2 w-full"
              src="https://www.interactivebrokers.hu/images/web/cryptocurrency-hero.jpg"
            />
          </div>
        </div>
      </div>
      <div
        class="app"
      >
        <div
          class="main"
        >
          <section
            class="ant-layout css-dev-only-do-not-override-8jqkd1"
          >
            <div
              class="routes"
            >
              <div
                class="loader"
              >
                <div
                  aria-busy="true"
                  aria-live="polite"
                  class="ant-spin ant-spin-spinning css-dev-only-do-not-override-8jqkd1"
                >
                  <span
                    class="ant-spin-dot ant-spin-dot-spin"
                  >
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                  </span>
                </div>
              </div>
            </div>
          </section>
          <div
            class="footer"
          >
            <h5
              class="ant-typography css-dev-only-do-not-override-8jqkd1"
              style="color: white; text-align: center;"
            >
              Copyright Â© 2021
              <a
                href="/"
              >
                Cryptoverse Inc.
              </a>
               
              <br />
              <p>
                All Rights Reserved.
              </p>
            </h5>
            <div
              class="ant-space css-dev-only-do-not-override-8jqkd1 ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <a
                  href="/"
                >
                  Home
                </a>
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <a
                  href="/cryptocurrencies"
                >
                  Cryptocurrencies
                </a>
              </div>
              <div
                class="ant-space-item"
              >
                <a
                  href="/news"
                >
                  News
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
